exonum/src/runtime/mod.rs

```diff
@@ -414,16 +420,16 @@ impl Caller {
 pub struct ExecutionContext<'a> {
     /// The current state of the blockchain. It includes the new, not-yet-committed, changes to
     /// the database made by the previous transactions already executed in this block.
-    pub fork: &'a Fork,
+    pub fork: &'a mut Fork,
     /// The initiator of the transaction execution.
     pub caller: Caller,
     /// Identifier of the service interface required for the call. Keep in mind that this field in
     /// fact is a part of unfinished "interfaces feature" and will be replaced in future releases.
     /// At the moment this field can only contains a core interfaces like `Configure` and
     /// always empty for the common the service interfaces.
     pub interface_name: &'a str,
-    /// Reference to the underlying runtime dispatcher.
-    dispatcher: &'a DispatcherRef<'a>,
+    /// Reference to the communication channel.
+    pub dispatcher: &'a mut Dispatcher,
```

Comment:
It's a bit strage how the idea of original PR was to encapsulate the dispatcher, and now it's even more public.

Reply:
My oversight; this field can be made private right away. There is still a `&mut Dispatcher` ref in `Runtime::after_commit`, though.